src/main/java/seedu/toluist/model/Task.java

```diff
@@ -61,4 +61,22 @@ public void replaceTags(ArrayList<Tag> tags) {
     public boolean isOverdue() {
         return startDateTime != null && startDateTime.isBefore(LocalDateTime.now());
     }
+
+    public boolean isStringContainedInDescriptionIgnoreCase(String comparison) {
+        return this.description.toLowerCase().contains(comparison.toLowerCase());
+    }
+
+    public boolean isStringContainedInAnyTagIgnoreCase(String comparison) {
+        for (Tag tag : allTags) {
+            if (tag.tagName.toLowerCase().contains(comparison.toLowerCase())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public int compareTo(Task comparison) {
```

Comment:
Duplicates occur cause when searching, multiple times the same result can occur e.g. search terms "homework" and "3" can both find "homework assignment 3" when typing `find homework 3`.